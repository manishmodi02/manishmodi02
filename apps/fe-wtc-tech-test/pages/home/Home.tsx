
import React, { FC } from 'react';
import { useMovies, MovieCard } from './index';
import {
    Container,
    Title,
    ListContainer,
    Grid
  } from './Home.styles';

/* Home component Using the custom hook */
export const Home: FC = () => {
    const { movies, updateMovieState } = useMovies();
  
    return (
      <Container>
        <Title data-testid="count-title">Movies {`(${movies.length})`}</Title>
        <ListContainer>
          <Grid>
            {movies.map((movie, index) => (
              <MovieCard
                data-testid="MovieCard-element"
                movie={movie}
                updateMovieState={updateMovieState}
                index={index}
              />
            ))}
          </Grid>
        </ListContainer>
      </Container>
    );
  };
  
  export default Home;