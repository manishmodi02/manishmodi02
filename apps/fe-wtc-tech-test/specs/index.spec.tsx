import React from 'react';
import { render, cleanup, screen, fireEvent } from '@testing-library/react';
import fetchMock from 'fetch-mock';

import Home, { MovieCard } from '../pages/home';

export const movieData = {
  Actors: 'Vivica A. Fox, Ambrosia Kelley, Michael Parks, James Parks',
  Awards: 'Nominated for 2 Golden Globes. Another 22 wins & 82 nominations.',
  BoxOffice: 'N/A',
  Country: 'USA',
  DVD: 'N/A',
  Director: 'Quentin Tarantino',
  Genre: 'Action, Crime, Thriller',
  Language: 'English, Cantonese, Mandarin, Spanish',
  Metascore: '83',
  Plot:
    'The Bride continues her quest of vengeance against her former boss and lover Bill, the reclusive bouncer Budd, and the treacherous, one-eyed Elle.',
  Poster:
    'https://m.media-amazon.com/images/M/MV5BNmFiYmJmN2QtNWQwMi00MzliLThiOWMtZjQxNGRhZTQ1MjgyXkEyXkFqcGdeQXVyNzQ1ODk3MTQ@._V1_SX300.jpg',
  Production: 'N/A',
  Rated: 'R',
  Ratings: [
    { Source: 'Internet Movie Database', Value: '8.0/10' },
    { Source: 'Rotten Tomatoes', Value: '84%' },
  ],
  Released: '16 Apr 2004',
  Runtime: '137 min',
  Saved: 'False',
  Title: 'Kill Bill: Vol. 2',
  Type: 'movie',
  Watched: 'True',
  Website: 'N/A',
  Writer:
    'Quentin Tarantino, Quentin Tarantino (character The Bride), Uma Thurman (character The Bride)',
  Year: '2004',
  imdbID: 'tt0378194',
  imdbRating: '8.0',
  imdbVotes: '657,553',
};

describe('test cases for home component', () => {
  beforeAll(() => {
    fetchMock.mock('http://localhost:3333/api/movies', {
      json: () => [movieData],
    });
    jest.spyOn(window, 'fetch');
  });

  beforeEach(() => {
    render(<Home />);
  });

  afterEach(() => {
    cleanup();
  });

  afterAll(() => {
    fetchMock.restore();
  });

  it('movies API should be called once', async () => {
    expect(fetch).toHaveBeenCalledTimes(1);
  });

  it('should render the homepage', () => {
    const HomeContainer = render(<Home />);
    expect(HomeContainer).toBeTruthy();
  });

  it('should match the snapshot', () => {
    const HomeComponent = render(<Home />);
    expect(HomeComponent).toMatchSnapshot();
  });

  it('should render Movies title', () => {
    const TitleNode = screen.getByTestId('count-title');

    expect(TitleNode).toBeTruthy();
  });
});

describe('test cases for movie card component', () => {
  let component;
  const defaultProps = {
    movie: movieData,
    index: 0,
    updateMovieState: jest.fn(),
  };

  beforeAll(() => {
    fetchMock.mock('http://localhost:3333/api/movies/id/tt0378194', {
      json: () => 'ok',
    });
    jest.spyOn(window, 'fetch');
  });

  beforeEach(() => {
    component = render(<MovieCard {...defaultProps}></MovieCard>);
  });

  afterEach(() => {
    cleanup();
  });

  afterAll(() => {
    fetchMock.restore();
  });

  it('should render movie card component', () => {
    expect(component).toMatchSnapshot();
  });

  it('should render movie card component', () => {
    expect(component).toBeTruthy();
  });

  it('should render view button container', () => {
    const ViewBtnElement = screen.getByTestId('view-btn-div');

    expect(ViewBtnElement).toBeTruthy();
  });

  it('should click on view eye button', () => {
    const ViewBtnElement = screen.getByTestId('view-btn-div');

    fireEvent.click(ViewBtnElement);

    // expect(fetch).toHaveBeenCalledTimes(1);
  });

  it('should render like button', () => {
    const LikeBtnElement = screen.getByTestId('like-btn-div');

    expect(LikeBtnElement).toBeTruthy();
  });

  it('should click on like button', () => {
    const LikeBtnNode = screen.getByTestId('like-btn-div');

    fireEvent.click(LikeBtnNode);
  });

  it('should render rating container', () => {
    const RatingContainer = screen.getByTestId('rating-container');

    expect(RatingContainer).toBeTruthy();
  });
});
