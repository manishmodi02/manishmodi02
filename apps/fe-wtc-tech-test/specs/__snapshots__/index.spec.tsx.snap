// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test cases for home component should match the snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-AxjAm bSjewG"
      >
        <h3
          class="sc-AxirZ cqPXHd"
          data-testid="count-title"
        >
          Movies 
          (0)
        </h3>
        <div
          class="sc-AxiKw bJIcTd"
        >
          <div
            class="sc-AxhCb kLExep"
          />
        </div>
      </div>
    </div>
    <div>
      <div
        class="sc-AxjAm bSjewG"
      >
        <h3
          class="sc-AxirZ cqPXHd"
          data-testid="count-title"
        >
          Movies 
          (0)
        </h3>
        <div
          class="sc-AxiKw bJIcTd"
        >
          <div
            class="sc-AxhCb kLExep"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-AxjAm bSjewG"
    >
      <h3
        class="sc-AxirZ cqPXHd"
        data-testid="count-title"
      >
        Movies 
        (0)
      </h3>
      <div
        class="sc-AxiKw bJIcTd"
      >
        <div
          class="sc-AxhCb kLExep"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`test cases for movie card component should render movie card component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-AxhUy eqwRKp"
      >
        <div
          class="sc-Axmtr hsLwGg"
        >
          <div
            data-testid="view-btn-div"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
              />
            </svg>
          </div>
          <div
            data-testid="like-btn-div"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M16.5 3c-1.74 0-3.41.81-4.5 2.09C10.91 3.81 9.24 3 7.5 3 4.42 3 2 5.42 2 8.5c0 3.78 3.4 6.86 8.55 11.54L12 21.35l1.45-1.32C18.6 15.36 22 12.28 22 8.5 22 5.42 19.58 3 16.5 3zm-4.4 15.55l-.1.1-.1-.1C7.14 14.24 4 11.39 4 8.5 4 6.5 5.5 5 7.5 5c1.54 0 3.04.99 3.57 2.36h1.87C13.46 5.99 14.96 5 16.5 5c2 0 3.5 1.5 3.5 3.5 0 2.89-3.14 5.74-7.9 10.05z"
              />
            </svg>
          </div>
        </div>
        <div
          class="sc-AxgMl fVVvGD"
        >
          <img
            class="sc-AxheI ieiRJK"
            src="https://m.media-amazon.com/images/M/MV5BNmFiYmJmN2QtNWQwMi00MzliLThiOWMtZjQxNGRhZTQ1MjgyXkEyXkFqcGdeQXVyNzQ1ODk3MTQ@._V1_SX300.jpg"
          />
        </div>
        <div
          class="sc-AxmLO kxoubb"
          data-testid="rating-container"
        >
          <text
            class="sc-fzozJi ehRjUM"
          >
            (8.0)
          </text>
          <div
            class="sc-fzoLsD fkGkWQ"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
              />
            </svg>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
              />
            </svg>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
              />
            </svg>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
              />
            </svg>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-AxhUy eqwRKp"
    >
      <div
        class="sc-Axmtr hsLwGg"
      >
        <div
          data-testid="view-btn-div"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
            />
          </svg>
        </div>
        <div
          data-testid="like-btn-div"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.5 3c-1.74 0-3.41.81-4.5 2.09C10.91 3.81 9.24 3 7.5 3 4.42 3 2 5.42 2 8.5c0 3.78 3.4 6.86 8.55 11.54L12 21.35l1.45-1.32C18.6 15.36 22 12.28 22 8.5 22 5.42 19.58 3 16.5 3zm-4.4 15.55l-.1.1-.1-.1C7.14 14.24 4 11.39 4 8.5 4 6.5 5.5 5 7.5 5c1.54 0 3.04.99 3.57 2.36h1.87C13.46 5.99 14.96 5 16.5 5c2 0 3.5 1.5 3.5 3.5 0 2.89-3.14 5.74-7.9 10.05z"
            />
          </svg>
        </div>
      </div>
      <div
        class="sc-AxgMl fVVvGD"
      >
        <img
          class="sc-AxheI ieiRJK"
          src="https://m.media-amazon.com/images/M/MV5BNmFiYmJmN2QtNWQwMi00MzliLThiOWMtZjQxNGRhZTQ1MjgyXkEyXkFqcGdeQXVyNzQ1ODk3MTQ@._V1_SX300.jpg"
        />
      </div>
      <div
        class="sc-AxmLO kxoubb"
        data-testid="rating-container"
      >
        <text
          class="sc-fzozJi ehRjUM"
        >
          (8.0)
        </text>
        <div
          class="sc-fzoLsD fkGkWQ"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
            />
          </svg>
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
            />
          </svg>
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
            />
          </svg>
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
            />
          </svg>
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
